src/cache.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#pragma once
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "list.h"
```

Comment:
Don't expose `list.h` in this public header. Encapsulate the details as possible.

Reply:
Then, it is feasible to switch among different cache replacement implementations later. Let's stick to consistent interface.